### PR TITLE
Type addition for postgres name, _json, _jsonb, and _timestamptz

### PR DESCRIPTION
**Pull request by mnemitz**
Added unit and integration tests for postgres name,  _json, _jsonb, and _timestamptz types.

### CODE REVIEW

**Comment by mnemitz** [comment]
Sorry not sure why the tests are failing on TravisCI, they pass locally and I updated every relevant file. Maybe tomorrow we could discuss it some more

**Comment by xiamx** [comment]
You might need to run the tests with this command to make it run with postgres database
`POSTGRES_URL="postgres://youruser@localhost/anyemptytestdatabase" npm test
`

File: test/fixture/postgres/osm.sql [hunk 1]
```diff
@@ -73,5 +73,9 @@ CREATE TABLE users (
     interval_col interval,
     json_col json,
     date_col date,
-    unspported_path_type path
+    unspported_path_type path,
+    name_type_col name,
+    json_array_col json[],
+    jsonb_array_col jsonb[],
+    timetamptz_array_col timestamptz[]
```
Comment (xiamx) [inline test/fixture/postgres/osm.sql]: looks like you have a typo over here.
CI logs: https://travis-ci.org/SweetIQ/schemats/jobs/277855858#L1482

Comment (mnemitz) [reply test/fixture/postgres/osm.sql]: Right you are, totally missed that! I'll change it now

**Comment by xiamx** [comment]
Release v3.0.1

**Pull request merged**